x-pack/functionbeat/docs/filtering.asciidoc

```diff
@@ -0,0 +1,41 @@
+[[filtering-and-enhancing-data]]
+== Filter and enhance the exported data
+
+Your use case might require only a subset of the data exported by {beatname_uc},
+or you might need to enhance the exported data (for example, by adding
+metadata). {beatname_uc} provides a couple of options for filtering and
+enhancing exported data.
+
+You can configure each input to include or exclude specific lines or files. This
+allows you to specify different filtering criteria for each input. To do this,
+you use the `include_lines`, `exclude_lines`, and `exclude_files` options under
+the +{beatname_lc}.inputs+ section of the config file (see
+<<configuration-{beatname_lc}-options>>). The disadvantage of this approach is
+that you need to implement a configuration option for each filtering criteria
+that you need.
```

Comment:
The above is not supported for functionbeat, users can define processors per function and use the drop events processors instead.

Reply:
Also in the context of cloudwatch its better to use a _filter_pattern_ that way the function is not invoked if the data doesn't match the pattern and reduce the cost of execution $$$ :)

Reply:
added to the docs, though it's not clear to me where this option is valid. You can comment when you review.